Homework4/ParseTree/ParseTree/ParseTree.cs

```diff
@@ -0,0 +1,121 @@
+namespace ParseTree;
+
+/// <summary>
+/// Class that represents parse tree data structure.
+/// </summary>
+public class ParseTree
+{
+    public ParseTree(string? expression)
+    {
+        if (expression == null)
+        {
+            throw new ArgumentNullException("Expression can't be null!");
+        }
+
+        if (expression == String.Empty)
+        {
+            throw new ArgumentException("Expression can't be empty!");
+        }
+
+        try
+        {
+            BuildTree(expression);
+        }
+        catch (ArgumentException)
+        {
+            throw;
+        }
```

Comment:
```suggestion
        BuildTree(expression);
```
Этот `try` ни на что не влияет. Получилась такая логика: если исключение `ArgumentException`, то упади с этим же исключением. При любом другом исключении, продолжай падать с этим же исключением.